### PR TITLE
Session log timestamps

### PR DESCRIPTION
**Pull request by Meatballs1**
Resolves: https://github.com/rapid7/metasploit-framework/issues/5215

Adds timestamps to session log files.

The methods `log_from_*` were not used anywhere, so I removed them, as they just confused me.

I have stripped out console colouring with a regex. I'm not sure where the ^A ^B characters are coming from?
## Verification
- [x] meterpreter logs contain timestamps and remain legible
- [x] shell logs contain timestamps and remain legible? 
## Example

Old meterpreter logs:

```
[*] Logging started: 2015-06-19 21:21:27 +0100

load stdapiload priv^A^[[4m^Bmeterpreter^A^[[0m^B > lsListing: C:\Users\User
======================

 Mode              Size    Type  Last modified              Name
 ----              ----    ----  -------------              ----
 40777/rwxrwxrwx   0       dir   2010-07-15 14:04:01 +0100  AppData
 40777/rwxrwxrwx   0       dir   2010-07-15 14:04:01 +0100  Application Data
```

New:

```
[06/19/2015 21:43:58] ^[[1m^[[34m[*]^[[0m Backgrounding session 1...

[06/19/2015 21:46:05] ^A^Bmeterpreter^A^B >·
[06/19/2015 21:46:05] ls
[06/19/2015 21:46:06] Listing: C:\Users\User
======================

Mode              Size    Type  Last modified              Name
----              ----    ----  -------------              ----
40777/rwxrwxrwx   0       dir   2010-07-15 14:04:01 +0100  AppData
40777/rwxrwxrwx   0       dir   2010-07-15 14:04:01 +0100  Application Data
40555/r-xr-xr-x   0       dir   2010-07-15 14:04:17 +0100  Contacts
40777/rwxrwxrwx   0       dir   2010-07-15 14:04:01 +0100  Cookies
```

### CODE REVIEW

**Comment by Meatballs1** [comment]
Not sure if new class is the way forwards. maybe should just add a method to log_dispatcher.

**Comment by Meatballs1** [comment]
I think session logging comes through rex::ui::text::shell.log_* methods. changing these to another method will change the way console is logged?

File: lib/rex/logging/sinks/timestamp_flatfile.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+###
+#
+# This class implements the LogSink interface and backs it against a
+# file on disk with a Timestamp.
+#
+###
+class TimestampFlatfile < Flatfile
+
+  def log(sev, src, level, msg, from) # :nodoc:
+    cleaned = msg.chop.gsub(/\x1b\[[0-9;]*[mG]/,'')
```
Comment (jhart-r7) [inline lib/rex/logging/sinks/timestamp_flatfile.rb]: what is this doing?

Comment (Meatballs1) [inline lib/rex/logging/sinks/timestamp_flatfile.rb]: removing terminal colours

File: lib/rex/logging/sinks/timestamp_flatfile.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    fd.write("[#{get_current_timestamp}] #{cleaned}\n")
+    fd.flush
+  end
+
+
```
Comment (wvu-r7) [inline lib/rex/logging/sinks/timestamp_flatfile.rb]: @MSadek-r7: Feel free to remove this extra newline when you land.

**Pull request merged**